### DISCUSSION TITLE
Adjacency matrix from bn model

### DISCUSSION THREAD

**Discussion by kenneth-lee-ch** [agrumery/aGrUM#21] [Q&A]
Hi, is there an efficient way to extract an adjacency matrix from a bn object where the columns of the matrix corresponds to the same order in the sample generated from the bn object?

**Comment by phwuil** [comment]
Hi @kenneth-lee-ch , there is no (particularly) efficient way to build this adjacency matrix since the internal representation is not this one. It is a simple code. The trickiest part is that you cannot rely on the nodeId to give the id for column or row in the matrix.

```python
def adjacencyMatrix(bn):
    n=bn.size()
    am=np.zeros((n,n))
    idmap={bn.topologicalOrder()[i]:i for i in range(n)}
    for i,j in bn.arcs():
        am[idmap[i],idmap[j]]=1
        #am[idmap[j],idmap[i]]=-1
    return am
```

![image](https://github.com/agrumery/aGrUM/assets/5980636/b932d899-7dee-461d-a9ca-7217d7c88bd9)

(here, I rely on the topological order to give the order of the columns/rows in the matrix)

![image](https://github.com/agrumery/aGrUM/assets/5980636/750ea269-656e-4220-b4f0-26a3d88a2b84)
